### DISCUSSION TITLE
Convert Image to Greyscale

### DISCUSSION THREAD

**Discussion by ab623** [hajimehoshi/ebiten#2879] [Q&A] [answered]
I have looked everywhere for an answer to this, apparently its not a problem that has ever existed, or its so easy that noone has asked a question on it!

I have the following code which I would have though would scale the image to greyscale (based on a standard scaling formula), however it come out with a green tint. I have tried multiple approaches but not found anything that seems to easily greyscale the image.

```go
func (s *screen) Draw(screen *ebiten.Image) {
	op := &ebiten.DrawImageOptions{}
	op.ColorScale.Scale(0.299, 0.587, 0.114, 1)

	screen.DrawImage(s.currentImage, op)
}
```

![image](https://github.com/hajimehoshi/ebiten/assets/3438974/e576fbaa-7fd0-43c1-b546-0baf3899fce0)

Any suggestions?

**Comment by hajimehoshi** [comment, marked as answer]
You can use `ColorM`.

https://pkg.go.dev/github.com/hajimehoshi/ebiten/v2/colorm#ColorM.ChangeHSV

```
op.ColorM.ChangeHSV(0, 0, 1)
```

See also `examples/hsv`

**Comment by ab623** [comment]
Thank you!

That has solved the issue. Although `ColorM` is being depreciated from `ebiten.DrawImageOptions`, so I am using the colorm packaged instead. 

For anyone in future that has the same issue and would like the see the updated code, I have pasted it below.

```go
func (s *screen) Draw(screen *ebiten.Image) {
	op := &colorm.DrawImageOptions{}

	var c colorm.ColorM
	c.ChangeHSV(0, 0, 1)

	colorm.DrawImage(screen, s.currentImage, c, op)
}
```
